confgenerator/otel/modular.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package otel provides data structures to represent and generate otel configuration.
+package otel
+
+import (
+	"fmt"
+
+	yaml "github.com/goccy/go-yaml"
+	"github.com/mitchellh/mapstructure"
```

Comment:
Does Otel also use `mapstructure`?

Another dumb question: What are the reasons we need `otel/modular.go` instead of using something Otel has? I guess

1. Otel does not already have something like this module?
2. Otel has something like this module, but it's not public?
2. We want to enforce rules that are specific to us (e.g. only one exporter. Always prefix with `metrics/` for pipelines.)?

Reply:
Got it